quesma/quesma/auth_middleware.go

```diff
@@ -0,0 +1,69 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package quesma
+
+import (
+	"net/http"
+	"quesma/elasticsearch"
+	"quesma/quesma/config"
+	"sync"
+	"time"
+)
+
+const cacheWipeInterval = 10 * time.Minute
+
+// authMiddleware a simple implementation of an authentication middleware,
+// which checks the Authorization header and validates it against Elasticsearch.
+//
+// If the validation is positive, the Authorization header is stored in a cache to avoid unnecessary calls to Elasticsearch preceding each request.
+// The cache is wiped every 10 minutes - all items at once, perhaps this could be revisited in the future.
+type authMiddleware struct {
+	nextHttpHandler   http.Handler
+	authHeaderCache   sync.Map
+	cacheWipeInterval time.Duration
+	esClient          elasticsearch.SimpleClient
+}
+
+func NewAuthMiddleware(next http.Handler, esConf config.ElasticsearchConfiguration) http.Handler {
+	esClient := elasticsearch.NewSimpleClient(&esConf)
+	middleware := &authMiddleware{nextHttpHandler: next, esClient: *esClient, cacheWipeInterval: cacheWipeInterval}
+	go middleware.startCacheWipeScheduler()
+	return middleware
+}
+
+func (a *authMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
This middleware is quiet. That's fine for production deployment.  It will be a PITA to solve any auth issues on the test/stg environment.  Maybe we should have a flag and log every unsuccessful attempt.

Reply:
I was actually thinking about bringing auth-related messaging to debug logs, WDYT?

Reply:
That's fine. I think we can run in debug mode in evaluation scenario.